reconcile/utils/ocm/addons.py

```diff
@@ -6,46 +6,209 @@
 from reconcile.utils.ocm.base import OCMAddonInstallation
 from reconcile.utils.ocm_base_client import OCMBaseClient
 
+ADDON_UPGRADE_POLICY_DESIRED_KEYS = {
+    "id",
+    "addon_id",
+    "schedule_type",
+    "schedule",
+    "next_run",
+    "version",
+}
 
-def get_addons_for_cluster(
-    ocm_api: OCMBaseClient,
-    cluster_id: str,
-    addon_latest_versions: dict[str, str],
-    required_state: Optional[str],
-) -> list[OCMAddonInstallation]:
-    """
-    Returns a list of Addons installed on a cluster
-
-    :param cluster_id: ID of the cluster
-    :param addon_latest_versions: dict of addon_id -> latest version. This allows us to
-        populate the addonsinstalation available upgrades in an efficient way.
-    :param required_state: only return addons with this state
-    """
-
-    params: Optional[dict[str, Any]] = None
-    if required_state:
-        params = {"search": f"state='{required_state}'"}
-
-    addons = []
-    for addon in ocm_api.get_paginated(
-        api_path=f"/api/clusters_mgmt/v1/clusters/{cluster_id}/addons",
-        params=params,
-    ):
-        current_version = addon["addon_version"]["id"]
-        latest_version = addon_latest_versions.get(addon["id"])
-        addon["addon_version"]["available_upgrades"] = (
-            [latest_version] if latest_version != current_version else []
+
+class AddonService:
```

Comment:
Why not make `ocm_api` an instance field? It's used in a lot of places.

Reply:
if we have access to `ocm_api` during class instanciation, it would make sense indeed

Reply:
i'm going to delete all this code after all OCM environments are on v2. so i wanted to keep the method signatures the same as they were before.